frontend/src/app/libs/auth-pages/sign-in/sign-in.component.ts

```diff
@@ -39,25 +55,30 @@ export class SignInComponent implements OnInit, OnDestroy {
 		});
 
 		this.signInFormGroup = this.formBuilder.group({
-			email: ['', [Validators.required, Validators.email]],
-			password: ['', [Validators.required, Validators.minLength(this.passwordMinLenth)]],
+			email: '',
+			password: '',
+			token: ''
 		});
 	}
 
-	public get emailInputErrorMessage(): string {
-		if ((this.signInFormGroupControl.email.touched || this.submitted) && !this.signInFormGroupControl.email.valid) {
-			const emailInputErrorMessage: string[] = Object.keys(this.signInFormGroupControl.email.errors);
-			return emailInputErrorMessage[0];
+	public emailValidator(control: AbstractControl): ValidationErrors | null {
+		const maxLength: number = 128;
+		const emailPattern: RegExp = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+		if (control && Boolean(control.value) && control.value.length >= maxLength ||
```

Comment:
Looks like we don't need to check twice on `control && Boolean(control.value)`

Reply:
I deleted redundant check.